docs/guides/tfhers.md

```diff
@@ -0,0 +1,340 @@
+
+   ```python
+   from concrete.fhe import tfhers
+
+   # don't worry about the API, we will have better examples later.
+   # we just want to show the encoding here
+   tfhers_params = tfhers.CryptoParams(
+   lwe_dimension=909,
+   glwe_dimension=1,
+   polynomial_size=4096,
+   pbs_base_log=15,
+   pbs_level=2,
+   lwe_noise_distribution=0,  # this won't be needed so we set it to 0
+   glwe_noise_distribution=2.168404344971009e-19,
+   encryption_key_choice=tfhers.EncryptionKeyChoice.BIG,
+   )
+
+   # TFHERSInteger using this type will be represented as a vector of 8/2=4 integers
+   tfhers_type = tfhers.TFHERSIntegerType(
+       is_signed=False,
+       bit_width=8,
+       carry_width=3,
+       msg_width=2,
+       params=tfhers_params,
+   )
+
+   # array([3, 2, 3, 1])
+   tfhers_type.encode(123)
+
+   # 123
+   tfhers_type.decode([3, 2, 3, 1])
+   ```
+
+2. The Concrete Optimizer can find parameters which are not in TFHE-rs pre-computed list of parameters. Therefore, we need to either fix or limit the search space in areas of the circuit where compatibility is required. This will ensure that the same parameters are used where needed.
+
+## First Use Case: Shared Secret Key
+
+The first scenario involves a single party seeking to combine both Concrete and TFHE-rs in a computation. In this case, a shared secret key will be used, while different keysets will be held for Concrete and TFHE-rs. It is possible to generate keys in both directions. In case 1.1, the steps are as follows:
+
+1. We do a classical keygen in Concrete, which generates a bunch of secret keys (we would be interested in the one used for TFHE-rs) and public keys
+2. We will use the secret key meant for TFHE-rs to launch a partial keygen (i.e. starting from a secret key and generating the rest of the keys) in TFHE-rs
+
+Case 2.2 is basically the opposite, the steps are as follows:
+
+1. We do a classical keygen in TFHE-rs, which generates a single secret key and different public keys
+2. We will use the secret key to launch a partial keygen in Concrete
+
+It is possible to generate keys in both directions, meaning that you can start by generating keys from Concrete and then use Concrete's secret key to generate the remaining keys (partial keygen) in TFHE-rs (use case 1.1), or vice versa (use case 1.2).
+
+After the key generation is complete and both keysets are acquired, we can perform computations, encryption, and decryption on both ends.
```

Comment:
I feel "both keysets are acquired," is not English. Aquired doesn't seem ok to me